src/runtime_src/core/edge/drm/zocl/zocl_sk.c

```diff
@@ -107,6 +107,13 @@ zocl_sk_create_ioctl(struct drm_device *dev, void *data, struct drm_file *filp)
 
 	mutex_lock(&sk->sk_lock);
 
+	if (cu_idx >= 128) {
```

Comment:
1. Use MAX_SOFT_KERNEL instead of 128
2. Maybe we should move this check to the first validation in this function so that we don't need to worry about BO refcount and mutex protect sk_cu